[Bug]: Drop support Angular versions

### Version

13.1.5

### Steps to reproduce

1. clone any repo with Angular 13 or 14
2. npm i jest-preset-angular@13.1.5

### Expected behavior

It should install everything without errors

### Actual behavior

PeerDependencies error:

> npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR!
npm ERR! While resolving: general-front@0.258.0
npm ERR! Found: @angular-devkit/build-angular@13.3.11
npm ERR! node_modules/@angular-devkit/build-angular
npm ERR!   dev @angular-devkit/build-angular@"~13.3.11" from the root project
npm ERR!
npm ERR! Could not resolve dependency:
npm ERR! peer @angular-devkit/build-angular@">=15.0.0 <18.0.0" from jest-preset-angular@13.1.5
npm ERR! node_modules/jest-preset-angular
npm ERR!   dev jest-preset-angular@"~13.1.2" from the root project
npm ERR!
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.


### Additional context

The semantic versioning is not being respected, a BREAKING CHANGE has been made in a patch version release

### Environment

```shell
System:
    OS: Windows 11 10.0.22000
    CPU: (8) x64 11th Gen Intel(R) Core(TM) i5-1135G7 @ 2.40GHz        
  Binaries:
    Node: 16.20.0 - C:\Program Files\nodejs\node.EXE
    npm: 8.19.4 - C:\Program Files\nodejs\npm.CMD
  npmPackages:
    jest: ~29.7.0 => 29.7.0
```

--- a/package.json
+++ b/package.json
@@ -55,10 +55,10 @@
     "esbuild": ">=0.13.8"
   },
   "peerDependencies": {
-    "@angular-devkit/build-angular": ">=15.0.0 <18.0.0",
-    "@angular/compiler-cli": ">=15.0.0 <18.0.0",
-    "@angular/core": ">=15.0.0 <18.0.0",
-    "@angular/platform-browser-dynamic": ">=15.0.0 <18.0.0",
+    "@angular-devkit/build-angular": ">=13.0.0 <18.0.0",
+    "@angular/compiler-cli": ">=13.0.0 <18.0.0",
+    "@angular/core": ">=13.0.0 <18.0.0",
+    "@angular/platform-browser-dynamic": ">=13.0.0 <18.0.0",
     "jest": "^29.0.0",
     "typescript": ">=4.4"
   },

